internal/controller/api/v1alpha1/installenv_types.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2020.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type InstallationEnvironmentSpec struct {
+	// Proxy defines the proxy settings for agents and clusters that use the
+	// InstallationEnvironment. If unset, the agents and clusters will not be 
+	// configured to use a proxy.
+	// +optional
+	Proxy *Proxy `json:"proxy,omitempty"`
+
+	// AdditionalNTPSources is a list of NTP sources (hostname or IP) to be added to 
+	// all cluster hosts.
+	// +optional
+	AdditionalNTPSources []string `json:"additionalNTPSources,omitempty"`
+
+	// SSHPublicKey is a list of SSH public key that will be added to all agents 
+	// for use in debugging.
+	// +optional
+	SSHPublicKey []string `json:"sshPublicKey,omitempty"`
+
+	// PullSecretRef is the reference to the secret to use when pulling images.
+	PullSecretRef *corev1.LocalObjectReference `json:"pullSecretRef"`
+
+	// Agents that boot as a result of this InstallationEnvironment should be labeled 
+	// such that they match this selector.
+	// That is how a user would identify which agents are associated with a particular 
+	// InstallationEnvironment.
+	AgentLabelSelector metav1.LabelSelector `json:"agentLabelSelector"`
+
+	// Any additional labels to apply to agents that are associated with this 
+	// InstallationEnvironment.
+	// +optional
+	AgentLabels []string `json:"agentLabels,omitempty"`
+
+	// ClusterDeploymentRef is the reference to the single ClusterDeployment that will 
+	// be installed from this InstallationEnvironment.
+	// Future versions will allow for multiple ClusterDeployments and this reference 
+	// will be removed.
+	ClusterDeploymentRef *corev1.LocalObjectReference `json:"clusterDeploymentRef,omitempty"`
```

Comment:
Per https://github.com/openshift/assisted-service/pull/861#discussion_r554096491 we have `ClusterReference` [here](https://github.com/openshift/assisted-service/blob/5cdc032bdfb646efaea59b3558edb0a387bb4f22/internal/controller/api/v1alpha1/cluster_types.go#L29). Should have also been used by image_types.go

Reply:
Fixing it here https://github.com/openshift/assisted-service/pull/978